src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -38,50 +39,101 @@ public ModelTypeProviderFields(InputModelType inputModel, TypeFactory typeFactor
             var parametersToFields = new Dictionary<string, FieldDeclaration>();
 
             var visitedMembers = new HashSet<ISymbol>(SymbolEqualityComparer.Default);
-
-            foreach (var inputModelProperty in inputModel.Properties)
+            foreach (var inputModelProperty in properties)
             {
-                var originalFieldName = inputModelProperty.Name.ToCleanName();
-                if (originalFieldName == inputModel.Name)
-                {
-                    originalFieldName = $"{originalFieldName}Property";
-                }
-                var propertyType = GetPropertyDefaultType(inputModel.Usage, inputModelProperty, typeFactory);
+                var originalFieldName = Configuration.Generation1ConvenienceClient
+                    ? inputModelProperty.Name == "null" ? "NullProperty" : inputModelProperty.Name.ToCleanName()
+                    : inputModelProperty.Name.ToCleanName();
+
+                originalFieldName = BuilderHelpers.DisambiguateName(modelName, originalFieldName);
```

Comment:
our previous code here is appending `Property` when the name of the property collides with the name of its containing type.
The new change calls `DisambiguateName` which will append `Value` to those properties. Is this OK?

Reply:
Here is an issue about this topic: https://github.com/Azure/autorest.csharp/issues/3851

Reply:
I haven't found any existing case in azure-sdk-for-net repo. Do you know any of those?

Reply:
well maybe not. This is kind of like a possibility discussion.

Reply:
I assume ApiCompat would've raised an error in the azure-sdk-for-net PR.